security.md

```diff
@@ -4,35 +4,39 @@ title: Reporting a Security Vulnerability
+Ballerina project maintainers take security issues very seriously and all the vulnerability reports are treated with the highest priority and confidentiality.
 
-We strongly encourage you to report security vulnerabilities to our private security mailing list: [security@ballerina.io](mailto:security@ballerina.io) - first, before disclosing them in any public forums.
+- [Reporting a vulnerability](#reporting-a-vulnerability)
+- [Handling a vulnerability](#handling-a-vulnerability)
 
-This is a private mailing list where only members of the WSO2 internal security team are subscribed to, and is treated as top priority.
+## Reporting a vulnerability
 
-If you wish to send secure messages to [security@ballerina.io](mailto:security@ballerina.io), you may use the following key:
+Ensure you are using the latest Ballerina version before you run an automated security scan or perform a penetration test against them.
 
-security@ballerina.io: 0168 DA26 2989 0DB9 4ACD  8367 E683 061E 2F85 C381 [pgp.mit.edu](https://pgp.surfnet.nl/pks/lookup?op=vindex&fingerprint=on&search=0xE683061E2F85C381)
+If you have any concerns regarding the security aspects of the source code or any other resource in this repo or have uncovered a security vulnerability, we strongly encourage you to report that to our private and highly confidential security mailing list: **security@ballerina.io** first using the below key without disclosing them in any forums, sites, or other groups - public or private. 
 
-## Vulnerability Information
+s0168 DA26 2989 0DB9 4ACD 8367 E683 061E 2F85 C381 pgp.mit.edu
 
-Please use the following template in reporting vulnerabilities:
+We will keep you informed of the progress towards a fix and disclosure of the vulnerability if reported issue is identified as a true positive. To protect the end-user security, these issues could be disclosed in other places only after it’s mitigation actions and disclosure process are completed.
+
+**Warning:** Please do not create GitHub issues for security vulnerabilities. Further, kindly refrain from sharing the vulnerability details you come across with other individuals. 
+
+Also, use the following template when reporting vulnerabilities so that it contains all the required information and helps expedite the analysis and mitigation process.
 
 - Vulnerable Ballerina artifacts(s) and version(s)
 - Overview: High-level overview of the issue and self-assessed severity
 - Description: Include the steps to reproduce
 - Impact: Self-assessed impact
 - Solution: Any proposed solution
 
-## Vulnerability Handling
+## Handling a vulnerability
+
+The below is an overview of the vulnerability handling process.
 
-Here's an overview of our vulnerability handling process:
+1. The user privately reports the vulnerability to security@ballerina.io. (The initial response time will be less than 24 hours).
```

Comment:
What is the reason behind changing the process? For ex: Initial response time has increased to 24 hours from 1 hour)

Reply:
Changed it as per [1].

[1] https://docs.wso2.com/display/Security/WSO2+Security+Vulnerability+Reporting+Guidelines

Reply:
The links highlighted by @praneesha shows process we follow for WSO2 products. 1 hour initial response is not practical for us in some rare situations (example: issues reported in IST night time might take more time).  

However, since security@ballerina.io is maintained separately, you may decide to use your own SLA here. Team may decide how to monitor the mailing list and respond to incidences. You can use WSO2 product processes described in [1] as a guideline.

Reply:
+1 to proceed with WSO2 guidelines for now. If we need to change we can discuss later and change.